src/cobra/manipulation/delete.py

```diff
@@ -57,113 +60,40 @@ def prune_unused_reactions(model: "Model") -> Tuple["Model", List["Reaction"]]:
-        Internal function that has outlived its purpose.
-
-    """
-    warn(
-        "The function `get_compiled_gene_reaction_rules` has outlived its purpose. "
-        "It will be removed soon.",
-        DeprecationWarning,
-    )
-    return {r: r.gpr for r in model.reactions}
-
-
-def find_gene_knockout_reactions(
+def knock_out_model_genes(
     model: "Model",
-    gene_list: List["Gene"],
-    compiled_gene_reaction_rules: Optional[Dict["Reaction", Module]] = None,
+    gene_list: Union[List["Gene"], Set["Gene"], List[str], Set[str]],
 ) -> List["Reaction"]:
-    """Identify reactions which will be disabled when genes are knocked out.
+    """Temporarily remove the effect of genes in `gene_list`.
+
+    It sets the bounds to "zero" for reactions catalysed by the genes in
+    `gene_list` if deleting the genes stops the reactions from proceeding.
+
+    The changes are reverted upon exit if executed within the model as context.
 
     Parameters
     ----------
     model: cobra.Model
-        The model for which to find gene knock-out reactions.
+        The model whose reaction bounds are to be set.
     gene_list: list of cobra.Gene
         The list of genes to knock-out.
-    compiled_gene_reaction_rules: dict of {reaction: compiled_string},
-                                  optional
-        If provided, this gives pre-compiled gene-reaction rule strings.
-        The compiled rule strings can be evaluated much faster. If a rule
-        is not provided, the regular expression evaluation will be used.
-        Because not all gene-reaction rule strings can be evaluated, this
-        dict must exclude any rules which can not be used with eval
-        (default None).
 
     Returns
     -------
-    list of cobra.Reaction
-       The list of cobra.Reaction objects which will be disabled.
-
-    .. deprecated:: 0.22.1
-        Internal function that has outlived its purpose.
-
+    reaction_list: list[cobra.Reaction]
+        A list of cobra.Reactions that had the bounds turned to zero.
     """
-    warn(
-        "The function `find_gene_knockout_reactions` has outlived its purpose. "
-        "It will be removed in the next minor version (0.23.0).",
-        DeprecationWarning,
-    )
-    potential_reactions = set()
+    orig_bounds = model.reactions.list_attr("bounds")
     for gene in gene_list:
         if isinstance(gene, str):
             gene = model.genes.get_by_id(gene)
-        potential_reactions.update(gene._reaction)
-    gene_set = {str(i) for i in gene_list}
-    if compiled_gene_reaction_rules is None:
-        compiled_gene_reaction_rules = {r: r.gpr for r in potential_reactions}
-
-    return [
-        r
-        for r in potential_reactions
-        if not compiled_gene_reaction_rules[r].eval(gene_set)
-    ]
+        gene.knock_out()
+    new_bounds = model.reactions.list_attr("bounds")
+    reaction_list = list()
+    for i, rxn in enumerate(model.reactions):
+        if orig_bounds[i] != new_bounds[i]:
```

Comment:
Exact comparison with floating point numbers is never a good idea.

Reply:
So math.isclose(), maybe with the solver defined tolerance?
Something else?
https://docs.python.org/3/whatsnew/3.5.html#pep-485-a-function-for-testing-approximate-equality

Reply:
Since numpy is a direct dependency you can also use `numpy.isclose` if you like. And I don't think the solver tolerance has an impact on bound equivalence unless I overlooked something.